### ISSUE TITLE
Better way of doing Authentication?

### ISSUE THREAD

**Issue by etsuo** [sakuraapi/api#113]
Right now there's a regex that excludes routes from needing authentication. For example:
```
export const authExcludedRoutes = [
  {
    method: {POST: true},
    regex: /^\/user/
  }
]
```
It might be nice to have something like this instead:
```
@Routable({
  authenticate: [AuthAudience]
  baseUrl: '/'
})
export class ConfigApi extends SakuraApiRoutable {

  // this route is authenticated by default
  @Route({
    method: 'delete',
    path: ''
  })
  async deleteHandler(req: Request, res: Response, next: NextFunction) {
    await this.configHandler(req, res);
    next();
  }

  @Route({
    method: 'get',
    path: '',
    authenticate: false
  })
  async getHandler(req: Request, res: Response, next: NextFunction) {
    await this.configHandler(req, res);
    next();
  }

  @Route({
    method: 'put',
    path: '',
    authenticate: [ServerAudience]
  })
  async putHandler(req: Request, res: Response, next: NextFunction) {
    await this.configHandler(req, res);
    next();
  }

  @Route({
    method: 'put',
    path: '',
    authenticate: [ServerAudience, AuthAnonymous]
  })
  async allowAnonymousHandler(req: Request, res: Response, next: NextFunction) {
    await this.configHandler(req, res);
    next();
  }
}

```
 
All routes will be subjected to verification through the `authAudience` plugin, except for `getHandler`.

`putHandler` will be subjected to verification through `serverAudience` then `authAudience`.

`allowAnonymousHandler` will be subjected to verification through `ServerAudience` but if authentication fails, the user will still be allowed through as anonymous.

**Comment by etsuo** [comment]
Depends on: https://github.com/sakuraapi/auth-audience/issues/6 (and vice versa)
